price-estimator/src/filter.rs

```diff
@@ -198,13 +184,12 @@ async fn estimate_buy_amount<T>(
         let token_info = get_token_info(token_pair.sell, token_infos.as_ref()).await?;
         sell_amount_in_quote * token_info.base_unit_in_atoms()
     };
-    let transitive_order = orderbook
-        .get_pricegraph()
-        .await
-        .order_for_sell_amount(token_pair, sell_amount_in_quote_atoms);
-    let buy_amount_in_base_atoms = transitive_order
-        .map(|order| apply_rounding_buffer(order.buy, price_rounding_buffer))
-        .unwrap_or_default();
+    let rounding_buffer = orderbook.rounding_buffer(token_pair).await;
+    let transitive_order = orderbook.pricegraph().await.order_for_sell_amount(
+        token_pair,
+        f64::max(sell_amount_in_quote_atoms - rounding_buffer, 0.0),
+    );
+    let buy_amount_in_base_atoms = transitive_order.map(|order| order.buy).unwrap_or_default();
```

Comment:
Before we were for first calling `order_for_sell_amount` then decreasing the buy amount of the resulting order.
Now we are first reducing the sell amount and then leaving the resulting order untouched.
We are doing it this way because this reduced sell amount is what the solver would see after applying the rounding buffer.